### PR TITLE
Fix capitalization in generated error messages

### PR DESCRIPTION
**Pull request by TW80000**
Eg: "Please enter a Phone Number" becomes "Please enter a phone number"

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]
Of course just works for the general case.  
Perhaps a ticket to capture future work to only lower case if only first letter is capitalized.

**Comment by TW80000** [comment]
Yeah, and we could be smarter about using 'an' instead of 'a' for labels that start with a vowel. But that opens up a whole new can of worms if we start thinking about i18n as well.

**Comment by kgrgreer** [comment]
You don't use "a" vs. "an" if the next word starts with a vowel or not, but if it sounds like a vowel.
Ex. An honour or A university.
Which means this would be very difficult to determine programmatically, and could even be different depending on the locale (not just language) of the user, depending on whether they pronounce things like leading H's or not.

File: src/foam/core/Validation.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ foam.CLASS({
         }
         return !required ? null : [[name],
           function() {
-            return !this.hasOwnProperty(name) && (`Please enter a ${label}`);
+            return !this.hasOwnProperty(name) && (`Please enter a ${label.toLowerCase()}`);
```
Comment (kgrgreer) [inline src/foam/core/Validation.js:105]: `Please provide a value for ${label.toLowerCase()}` avoids the a vs an problem.

File: src/foam/core/Validation.js [hunk 2]
```diff
@@ -129,7 +129,7 @@ foam.CLASS({
             predicateFactory: function(e) {
               return e.GTE(foam.mlang.StringLength.create({ arg1: self }), self.minLength);
             },
-            errorString: `Please enter a ${this.label} with least ${this.minLength} character${this.minLength>1?'s':''}`
+            errorString: `Please enter a ${this.label.toLowerCase()} with least ${this.minLength} character${this.minLength>1?'s':''}`
```
Comment (kgrgreer) [inline src/foam/core/Validation.js:132]: Same as above, here and below.

File: src/foam/core/Validation.js (hunk 1 above)
Comment (TW80000) [reply src/foam/core/Validation.js:105]: Even simpler is "This field cannot be blank" or "This field is required". What do you think about that instead?

Comment (jlhughes) [reply src/foam/core/Validation.js]: required fields will have a red boarder and red star underneath with this text. 

> * required
 
I think is sufficient. No issues with locale, property names. ... 
The placeholder text can handle provide a template/default/example.

Comment (jlhughes) [reply src/foam/core/Validation.js:105]: required fields will have a red boarder and red star underneath with this text. 

> * required
 
I think this is sufficient. No issues with locale, property names. ... 
The placeholder text can handle providing a template/default/example.

File: src/foam/core/Validation.js (hunk 2 above)
Comment (jlhughes) [reply src/foam/core/Validation.js:132]: also if sticking with the remainder then 
with least -> with at least

**Comment by TW80000** [comment]
This requires more thought, closing.

**Pull request closed without merging**